ssl/statem/statem_srvr.c

```diff
@@ -723,7 +730,7 @@ WORK_STATE ossl_statem_server_pre_work(SSL *s, WORK_STATE wst)
         return WORK_FINISHED_CONTINUE;
 
     case TLS_ST_SW_SESSION_TICKET:
-        if (SSL_IS_TLS13(s) && s->sent_tickets == 0) {
+        if (SSL_IS_TLS13(s) && s->sent_tickets == 0 && s->ext.extra_tickets_expected == 0) {
```

Comment:
This check worries me a bit, since it implicitly relies on the invariant that `s->ext.extra_tickets_expected` is zero during the initial handshake.  (A potential design to implement "only send tickets on first application data" on top of this would violate that invariant.)

Reply:
Perhaps an ossl_assert here to check this invariant is appropriate?

Reply:
I don't think I can actually assert anything at this location, since (as mentioned above) `SSL_IS_FIRST_HANDSHAKE()` is false already by this point.  Preventing calls to `SSL_new_session_ticket()` should be enough, though if we do end up adding an option to defer the initial batch of tickets until the first write we'll need to check whether `sent_tickets` should be updated for all tickets (not just initial ones)